metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py

```diff
@@ -536,11 +537,17 @@ def from_dict(
         resolved_includes: List[ProjectInclude],
         looker_viewfile_loader: "LookerViewFileLoader",
         reporter: "LookMLSourceReport",
+        model_explores_map: Dict[str, dict],
     ) -> "LookerExplore":
-        view_names = set()
+
+        view_names: Set[str] = set()
         joins = None
-        # always add the explore's name or the name from the from clause as the view on which this explore is built
-        view_names.add(dict.get("from", dict.get("name")))
+        assert "name" in dict, "Explore doesn't have a name field, this isn't allowed"
```

Comment:
asking to learn: if someone made a lookml like this, why isn't it allowed?

Reply:
would this lookml also not be accepted by looker or is a datahub specific requirement?

Reply:
the lkml parser will populate the `name` field in the dict based on the explore definition, that is the contract.
So, it's not about having a name field inside the explore definition... more like parsing the explore will populate the name field. 

e.g.
```
explore: events {
  ... explore def
}
```
will result in a dictionary with the `name` key set to `events`

Reply:
ah I see, so we'd be looking at a malformed explore then if this wasn't parsed